src/sidebar/components/test/markdown-editor-test.js

```diff
@@ -301,6 +305,215 @@ describe('MarkdownEditor', () => {
+      assert.isTrue(document.activeElement.innerText.indexOf(partialText) >= 0);
+    };
+
+    /**
+     * Asserts there should only be one "0" `tabIndex` value at a time which
+     * should be set on the focused element. All other `tabIndex` values
+     * on elements shall be "-1".
+     */
+    const testRovingIndex = () => {
+      assert.isTrue(document.activeElement.getAttribute('tabIndex') === '0');
+      assert.equal(
+        wrapper.find('ToolbarButton[tabIndex=0]').length +
+          wrapper.find('a[tabIndex=0]').length,
+        1
+      );
+    };
+
+    context('when `isPreviewing` is false', () => {
+      it('changes focus circularly to the left', () => {
+        pressKey('ArrowLeft');
+        // preview is the last button
+        matchesFocusedText('Preview');
+        testRovingIndex();
+      });
+
+      it('changes focus circularly to the right', () => {
+        pressKey('ArrowLeft'); // move to the end node
+        pressKey('ArrowRight'); // move back to the start
+        assert.isTrue(
+          document.activeElement.getAttribute('title').indexOf('Bold') >= 0
+        );
+        testRovingIndex();
+      });
+
+      it('changes focus to the last element when pressing `end`', () => {
+        pressKey('End'); // move to the end node
+        matchesFocusedText('Preview');
+        testRovingIndex();
+      });
+
+      it('changes focus to the first element when pressing `home`', () => {
+        pressKey('ArrowRight'); // move focus off first button
+        pressKey('Home');
+        matchesFocusedTitle('Bold');
+        testRovingIndex();
+      });
+
+      it('preserves the elements order and roving index', () => {
+        [
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Italic');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Quote');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Insert link');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Insert image');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Insert math (LaTeX is supported)');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Numbered list');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Bulleted list');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedTitle('Formatting help');
+            },
+          },
+          {
+            resultFn: () => {
+              matchesFocusedText('Preview');
+            },
+          },
+          {
+            resultFn: () => {
+              // back to the start
+              matchesFocusedTitle('Bold');
+            },
+          },
```

Comment:
Just as a suggestion, now that this is simplified you could streamline the array a bit further by making each item an object with `title` or `text` properties and then calling `matchesFocusedTitle` or `matchesFocusedText` accordingly:

```js
[...,
  { text: 'Preview' },
  { title: 'Bold' }
].forEach(({ text, title )) => {
...
}
```

The advantage is just that it saves a few lines and makes the whole test quicker to read.

Reply:
great suggestion, thanks